File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/geom/RJSLineString.java
package fr.sncf.osrd.railjson.schema.geom;

import java.util.ArrayList;
import java.util.List;

public class RJSLineString {

    public String type;

    public List<List<Double>> coordinates;

    public RJSLineString(String type, List<List<Double>> coordinates) {
        this.type = type;
        this.coordinates = coordinates;
    }

    /** Instantiates a line string from xs and ys coordinates */
    public static RJSLineString make(List<Double> xs, List<Double> ys) {
        assert (xs.size() == ys.size());
        var res = new ArrayList<List<Double>>();
        for (int i = 0; i < xs.size(); i++) {
            res.add(List.of(xs.get(i), ys.get(i)));
        }
        return new RJSLineString("LineString", res);
    }
}

File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/geom/RJSLineString.kt
package fr.sncf.osrd.railjson.schema.geom

class RJSLineString(var type: String, var coordinates: List<List<Double>>) {
    companion object {
        /** Instantiates a line string from xs and ys coordinates */
        fun make(xs: List<Double>, ys: List<Double>): RJSLineString {
            assert(xs.size == ys.size)
            val res = mutableListOf<List<Double>>()
            for (i in xs.indices) {
                res.add(listOf(xs.get(i), ys.get(i)))
            }
            return RJSLineString("LineString", res)
        }
    }
}
